test/nodes/test_file_converter.py

```diff
@@ -184,6 +185,35 @@ def test_pdf_parallel_sort_by_position(Converter):
     assert pages[-1] == "This is the page 50 of the document."
 
 
+@pytest.mark.integration
+@pytest.mark.parametrize("Converter", [PDFToTextConverter])
+def test_pdf_parallel_ocr(Converter):
+    converter = Converter(multiprocessing=True, sort_by_position=True, ocr="full", ocr_language="eng")
+    document = converter.convert(file_path=SAMPLES_PATH / "pdf" / "sample_pdf_6.pdf")[0]
+
+    pages = document.content.split("\f")
+
+    assert pages[0] == "This is the page 1 of the document."
+    assert pages[-1] == "This is the page 50 of the document."
+
+
+@pytest.mark.unit
+@fail_at_version(1, 17)
+def test_deprecated_ocr_node():
+    class MockPDFToTextOCRConverter:
+        def __init__(self) -> None:
+            warnings.warn(
+                """
+                The PDFToTextOCRConverter node is deprecated and will be removed in future versions.
+                Please use the PDFToTextConverter node instead and set the parameter ocr and ocr_language.
+                """,
+                category=DeprecationWarning,
+            )
+
+    with pytest.warns(DeprecationWarning):
+        converter = MockPDFToTextOCRConverter()
```

Comment:
```suggestion
@pytest.mark.unit
@fail_at_version(1, 17)
@patch("haystack.nodes.file_converter.image.ImageToTextConverter.__new__")
def test_deprecated_ocr_node(mock):
    with pytest.warns(DeprecationWarning):
        PDFToTextOCRConverter()
```

You'll need to add `from unittest.mock import patch` too.

Testing the mock you just created is not that useful, I know it's just to have the warning but testing the real class is better.